draft-ietf-mls-protocol.md

```diff
@@ -910,23 +910,34 @@ struct {
     opaque signature_key<0..2^16-1>;
 } BasicCredential;
 
+struct {
+    opaque cert_data<0..2^16-1>;
+} Certificate;
+
 struct {
     CredentialType credential_type;
     select (Credential.credential_type) {
         case basic:
             BasicCredential;
 
         case x509:
-            opaque cert_data<1..2^24-1>;
+            Certificate chain<1..2^32-1>;
     };
 } Credential;
 ~~~~~
 
 A BasicCredential is a raw, unauthenticated assertion of an identity/key
-binding.  The format of the key in the `public_key` field is defined by the
+binding. The format of the key in the `public_key` field is defined by the
 relevant ciphersuite: the group ciphersuite for a credential in a ratchet tree,
 the KeyPackage ciphersuite for a credential in a KeyPackage object.
 
+For X509Credential, each entry in the chain represents a single DER-encoded 
+X509 certificate. The chain is ordered such that the first entry (chain[0]) 
+is the end-entity certificate and each subsequent certificate in the chain 
+MUST be the issuer of the previous certificate. The algorithm for the  
+`public_key` in the end-entity certificate MUST match the relevant 
```

Comment:
This wasn't clearly defined before so I appreciate making it clear.
However, this `MUST` makes the mandatory ciphersuite pretty hard to use with x509 certificates. Public CAs don't issue ed25519 certificates.

Reply:
Most uses for MLS don’t involve public CAs.

Reply:
It seems a bit early to talk about "most uses for MLS", since AFAIK there aren't any yet :) 

But I think the overall point still stands -- precedent in the Web PKI doesn't really matter here, because (again in most of the envisioned use cases), these certificates will be for user identifiers that aren't even allowed in Web PKI certificates.  I suppose it's possible to have a bunch of entities with domain names and Web PKI certificates doing MLS, but (a) that doesn't seem like a core use cases and (b) they can just use a ciphersuite that is lines up with those certs.